pubsub/google/cloud/pubsub/client.py

```diff
@@ -236,3 +238,48 @@ def topic(self, name, timestamp_messages=False):
         :returns: Topic created with the current client.
         """
         return Topic(name, client=self, timestamp_messages=timestamp_messages)
+
+    def subscription(self, name, ack_deadline=None, push_endpoint=None,
+                     retain_acked_messages=None,
+                     message_retention_duration=None):
+        """Creates a subscription bound to the current client.
+
+        Example:
+
+        .. literalinclude:: pubsub_snippets.py
+           :start-after: [START client_subscription]
+           :end-before: [END client_subscription]
+           :dedent: 4
+
+        :type name: str
+        :param name: the name of the subscription to be constructed.
+
+        :type ack_deadline: int
+        :param ack_deadline: (Optional) The deadline (in seconds) by which
+                             messages pulledfrom the back-end must be
+                             acknowledged.
+
+        :type push_endpoint: str
+        :param push_endpoint:
+            (Optional) URL to which messages will be pushed by the back-end.
+            If not set, the application must pull messages.
+
+        :type retain_acked_messages: bool
+        :param retain_acked_messages:
+            (Optional) Whether to retain acked messages. If set, acked messages
+            are retained in the subscription's backlog for a duration indicated
+            by ``message_retention_duration``.
+
+        :type message_retention_duration: :class:`datetime.timedelta`
+        :param message_retention_duration:
+            (Optional) Whether to retain acked messages. If set, acked messages
+            are retained in the subscription's backlog for a duration indicated
+            by ``message_retention_duration``. If unset, defaults to 7 days.
+
+        :rtype: :class:`~google.cloud.pubsub.subscription.Subscription`
+        :returns: Subscription created with the current client.
+        """
+        return Subscription(
+            name, ack_deadline=ack_deadline, push_endpoint=push_endpoint,
+            retain_acked_messages=retain_acked_messages,
+            message_retention_duration=message_retention_duration, client=self)
```

Comment:
@kir-titievsky I wasn't sure if I should be allowing an optional `topic` to pass through here? (Or a topic name?)

As I mentioned in #3369, **without** a topic, some `Subscription` operations will fail (AFAIK it's only `Subscription.create()` but my check wasn't the most thorough check ever done).